docs/src/pages/demos/selects/NativeSelects.tsx

```diff
@@ -0,0 +1,228 @@
+import React from 'react';
+import { makeStyles, Theme } from '@material-ui/core/styles';
+import Input from '@material-ui/core/Input';
+import OutlinedInput from '@material-ui/core/OutlinedInput';
+import FilledInput from '@material-ui/core/FilledInput';
+import InputLabel from '@material-ui/core/InputLabel';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import NativeSelect from '@material-ui/core/NativeSelect';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  root: {
+    display: 'flex',
+    flexWrap: 'wrap',
+  },
+  formControl: {
+    margin: theme.spacing(1),
+    minWidth: 120,
+  },
+  selectEmpty: {
+    marginTop: theme.spacing(2),
+  },
+}));
+
+function NativeSelects() {
+  const classes = useStyles();
+  const [state, setState] = React.useState<{ age: string | number; name: string }>({
+    age: '',
+    name: 'hai',
+  });
+
+  const inputLabel = React.useRef<HTMLLabelElement>(null);
+  const [labelWidth, setLabelWidth] = React.useState(0);
+  React.useEffect(() => {
+    setLabelWidth(inputLabel.current!.offsetWidth);
```

Comment:
We can be pragmatic here I suppose.

Reply:
Do you think it would be better to have it split into an interface?

Reply:
I meant the non-null assertion. There is a subtle issue with this approach but that's not related to typescript.